### PR TITLE
Django 1.7 support

### PR DESCRIPTION
**Pull request by rlr**
To test this, you need to.

1- create new virtual env
2- `$ pip install fabric`
3- `$ pip install django==1.7.5`
4- `$ pip install -r requirements-test.txt`
5- `$ fab test`

You can also `$ pip install django==1.6.10` and retest. There is one existing failure already with django 1.6 that I vaguely remember punting on the last time we were in here. No failures with 1.7 though :facepunch: 

r?

### CODE REVIEW

**Comment by willkg** [comment]
We just chatted in IRC, but for public consumption:

1. I'd set up a tox environment for Django 1.7.
2. How does the GenericRelation change affect things?

I think other than that, this looks ok. I'll pull it down and run it through tox.

**Comment by rlr** [comment]
:point_up: added django 1.7 to tox.ini

**Comment by rlr** [comment]
That `GenericRelation` change would only affect ORM queries trying to go from the Watch->notifications. I'm not sure why it was there but am counting on things to be OK since tests pass.

File: tox.ini [hunk 1]
```diff
@@ -42,6 +42,15 @@ commands = pip install "Django<1.6.99"
            pip install -r requirements-test.txt
            {envpython} test_app/manage.py test tidings
 
+[testenv:py27_dj17]
+basepython = python2.7
+commands = pip install "Django<1.7.99"
+           pip install .
+           {envpython} test_app/manage.py test tidings
```
Comment (willkg) [inline tox.ini]: I'm pretty sure you don't need this line and the line above it. It causes this testenv to run the tests twice which I don't think is needed.

File: tox.ini [hunk 2]
```diff
@@ -42,6 +42,15 @@ commands = pip install "Django<1.6.99"
...
+           pip install -e .
+           pip install -r requirements-test.txt
+           {envpython} test_app/manage.py test tidings
+
 [testenv:sumo]
 basepython = python2.6
 commands = pip install "Django<1.5.99" "celery==3.0.24" "django-celery==3.0.23"
```
Comment (willkg) [inline tox.ini]: You might want to update the sumo environment to match what sumo has now.

**Comment by willkg** [comment]
I think everything looks ok here.

@erikrose Want to weigh in?

**Comment by rlr** [comment]
I updated tox.ini with kitsune's current requirements.

**Comment by rlr** [comment]
@erikrose said on IRC that he trusts our decisions. YAY!

**Pull request merged**